backend/routes/api/memberships.js

```diff
@@ -141,4 +135,115 @@ router.post("/:groupId", requireAuth, async (req, res, next) => {
+        err.status = 400
+        err.errors = { status: "Cannot change a membership status to pending"}
+        return next(err)
+    }
+
+    const coHost = await Membership.findOne({where: { memberId: req.user.id, groupId: groupId }})
+    const member = await Membership.findOne({where: { memberId: memberId, groupId } })
+
+    // check if user is a member
+    if (!member) {
+        const err = new Error('Validation Error')
+        err.status = 400
+        err.errors = { status: "User couldn't be found"}
+        return next(err)
+    }
+
+    if (member.status === status) {
+        const err = new Error('Validation Error')
+        err.status = 400
+        err.errors = { status: "User already have that status"}
+        return next(err)
+    }
+
+    // checks if belong to that group
+    if (member.groupId !== group.organizerId) {
+        const err = new Error("Membership between the user and the group does not exists")
+        err.status = 400
+        return next(err)
+    }
+
+    if ((coHost && coHost.status === 'co-host') || (req.user.id === group.organizerId)) {
+
+        if (member.status === 'pending') {
+            await member.update({ memberId, status})
+            const safeStatus = { id: member.id, groupId, memberId, status }
+            res.status(200).json(safeStatus)
+        }
+
+        if (req.user.id === group.organizerId && member.status === 'member' && status === 'co-host') {
+            await member.update({ memberId, groupId, status })
+            const safeStatus = { id: member.id, groupId, memberId, status }
+            res.status(200).json(safeStatus)
+        }
+    } else {
+        const err = new Error('User does not have permission')
+        err.status = 403
+        return next(err)
+    }
+});
+
+// ---------------- DELETE ENDPOINTS -------------------------
+
+router.delete('/:groupId', requireAuth, async (req, res, next) => {
+    const { groupId } = req.params
+    const { memberId } = req.body
+
+    const group = await Group.findOne({ where: { id: groupId }})
+    const user = await User.findOne({where: {id: memberId}})
+    const member = await Membership.findOne({ where: { memberId: memberId, groupId: group.organizerId}})
+
+    // throw error if group does not exist
+    if (!group) {
+        const err = new Error("Group couldn't be found")
+        err.status = 404
+        return next(err)
+    }
+
+    // if user does not exist
+    if(!user) {
+        const err = new Error("Validation Error")
+        err.status = 400
+        err.errors = { memberId: "User couldn't be found"}
+        return next(err)
+    }
+
+    // throw error if they are not a member
+    if(!member) {
+        const err = new Error("Membership does not exist for this User")
+        err.status = 404
+        return next(err)
+    }
+
+    // if user is organizer or user, delete membership
+    if (req.user.id === group.organizerId || member.groupId === group.organizerId) {
```

Comment:
second condition doesn't match the comment. 

compare current user id to organizer id ( done )
compare current user id to the member id for the second condition